operations.rb

```diff
@@ -5,116 +5,145 @@
+    elsif type == 2
+      specialization = create_teacher
+      new_person = Teacher.new(age, specialization, name: name)
+    else
+      puts 'Invalid option, try again.'
+      create_person
+      return
+    end
+
+    @people << new_person
+    puts 'Person created successfully'
   end
 
-  people << new_person
-  puts 'Person created successfully'
-end
+  # option 4
+  def create_book
+    puts 'Enter the title:'
+    title = gets.chomp
+    puts 'Enter the author:'
+    author = gets.chomp
+    new_book = Book.new(title, author)
+    @books << new_book
+    puts 'Book created successfully'
+  end
 
-# option 4
+  # option 5
+  def create_rental
+    puts 'Select a book from the following list by number'
+    @books.each_with_index { |book, index| puts "#{index}) #{book.title}" }
+    book_index = gets.chomp.to_i
 
-def create_book(books)
-  puts 'Enter the title:'
-  title = gets.chomp
-  puts 'Enter the author:'
-  author = gets.chomp
-  new_book = Book.new(title, author)
-  books << new_book
-  puts 'Book created successfully'
-end
+    puts 'Select a person from the following list by number'
+    @people.each_with_index { |person, index| puts "#{index}) #{person.name}" }
+    person_index = gets.chomp.to_i
 
-# option 5
-def create_rental(books, people)
-  puts 'Select a book from the following list by number'
-  books.each_with_index { |book, index| puts "#{index}) #{book.title}" }
-  book_index = gets.chomp.to_i
+    puts "Enter the date in format 'YYYY-MM-DD':"
+    date = gets.chomp
 
-  puts 'Select a person from the following list by number'
-  people.each_with_index { |person, index| puts "#{index}) #{person.name}" }
-  person_index = gets.chomp.to_i
+    Rental.new(date, @books[book_index], @people[person_index])
 
-  puts "Enter the date in format 'YYYY-MM-DD':"
-  date = gets.chomp
+    puts 'Rental created successfully'
+  end
 
-  Rental.new(date, books[book_index], people[person_index])
+  # option 6
+  def list_rentals_for_person_id
+    puts 'Enter the person id:'
+    id = gets.chomp.to_i
 
-  puts 'Rental created successfully'
-end
+    person = @people.find { |p| p.id == id }
 
-# option 6
-def list_rentals_for_person_id(people)
-  puts 'Enter the person id:'
-  id = gets.chomp.to_i
+    if person.nil?
+      puts "No person found with id #{id}"
+      return
+    end
 
-  person = people.find { |p| p.id == id }
+    person.rentals.each do |rental|
+      puts "Date: #{rental.date}, #{rental.book.title} rented by #{person.name}"
+    end
+  end
 
-  if person.nil?
-    puts "No person found with id #{id}"
-    return
+  def exit_program
+    puts 'Goodbye!'
+    exit
   end
 
-  person.rentals.each do |rental|
-    puts "Date: #{rental.date}, #{rental.book.title} rented by #{person.name}"
+  def execute_option(option)
+    if ACTIONS.key?(option)
+      send(ACTIONS[option])
+    else
+      puts 'Invalid selection. Please choose a valid option.'
+    end
   end
 end
```

Comment:
- [ ] As per the project requirement, you should pay special attention to the "Single-responsibility Principle" but here is the case that your `LibraryOperations` class does violate the Single-Responsibility Principle (SRP). The class is responsible for handling various operations related to the library, such as listing books and people, creating new people (both students and teachers), creating books, creating rentals, and listing rentals for a person's ID. This is more than one responsibility and is a clear violation of the SRP.
  
  The responsibilities of the LibraryOperations class can be broken down into the following:
    
    1. Managing interactions with books and book-related operations (listing books, creating books). :relaxed: 
    2. Managing interactions with people and person-related operations (listing people, creating people - students and teachers). :relaxed: 
    3. Managing interactions with rentals and rental-related operations (creating rentals, listing rentals for a person ID). :relaxed: 
  
  A better approach would be to refactor the class into smaller, more focused classes, each responsible for one of these responsibilities. For example, you could have separate classes like `BookManager`, `PersonManager`, and `RentalManager` to handle their respective operations. This way, each class adheres to the Single-Responsibility Principle, making the codebase more maintainable, modular, and easier to understand. :blush: